---
title: About
date: "2020-10-22T00:00:00.000Z"
description: "About page."
---

**EPI2ME Labs** maintains a collection of example bioinformatics notebooks for
running analysis in your web browser, and bioinformatics workflows and resources
for running automatically bioinformatics analyses at scale.

Our Github repositories contain a growing collection of resources on a range of
topics from basic quality control to genome assembly. These are free and open
to use by anyone.


## Nextflow workflows

The [epi2me-labs](https://github.com/epi2me-labs) Github space contains a
nascent collection of bioinformatics workflows written in Nextflow and tailored
to Oxford Nanopore Technologies' sequencing data.

### Getting started

All bioinformatics workflows available through [epi2me-labs](https://github.com/epi2me-labs)
are named consistently with the `wf-` prefix. They all contain instructions
in their `README` for installation and use.

* The [Workflows Quick Start Guide](/wfquickstart) contains additional information.
* Our [Index of Workflows](/wfindex) contains a listing of our currently available
  workflows and examples of their outputs.


## EPI2ME Labs Notebooks

Our Python notebook [Tutorials](https://github.com/epi2me-labs/tutorials)
walk through various topics suitable for bioinfomatics beginners to virtuosos.

EPI2ME Labs extends the JupyterLab notebook framework with a pre-configured
analysis server. The notebooks contain Python code needed to run data analysis,
however zero configuration of the code is required. The notebooks can be used
both by **students** and **researchers** new to nanopore sequence data
analysis, and more experienced bioinformaticians who are comfortable working
with Python.

The EPI2ME Labs server runs in a [Docker
container](https://www.docker.com/resources/what-container) to provide an
isolated and resettable environment.

### Getting started

EPI2ME Labs notebooks are supported on the GridION Mk1 sequencing platform as
well as any Windows 10, macOS, or Linux desktop. All components are also
released open-source in our [GitHub](https://github.com/epi2me-labs)
repository.

* The [Quick Start guide](/quickstart) will help you to get started running a
  notebook server and web-based analyses.
* Our [Index of Notebooks](/nbindex) page has a complete listing of our available
  notebooks with brief descriptions.
* The [Downloads](/downloads) page contains a list of all available resources to
  get you started.

To learn more about the Jupyter project, see
[jupyter.org](https://www.jupyter.org/).
